docs/REST-API/18-PCL-Overview.md

```diff
@@ -0,0 +1,590 @@
+
+There are two basic types of evaluation warnings: non-boolean evaluations and boolean results with evaluation warnings. 
+
+### Non-Boolean Evaluation
+
+[Non-Boolean Evaluation]:#non-boolean-evaluation
+
+These are PCL sub-expressions where the results are ambiguous / non-sensical. A classic example is an invalid type comparison like `2 > 'two'`. 
+
+PCL treats non-boolean evaluation as a `false` [boolean response with a warning][Boolean Evaluation With Warnings].
+
+Expression | Evaluates like | Result | Warning
+-- | -- | -- | --
+`2 > 'two'` | `non-boolean evaluation` | ❌ `false` | Type mismatch: `>` requires a [number] or [datetime] on both sides but got `[number] > [string]`
+`not 2 > 'two'` | `not (non-boolean evaluation)` -> `not (false)` -> | ✅ `true` | Type mismatch: `>` requires a [number] or [datetime] on both sides but got `[number] > [string]`
+`2 >= 'two' or 2 < 10` | `(non-boolean evaluation) or true` -> `false or true` | ✅ `true` | Type mismatch: `>=` requires a [number] or [datetime] on both sides but got `[number] >= [string]`
+`2 <= 'two' and 2 < 10` | `(non-boolean evaluation) and true` -> `false and true` | ❌ `false` | Type mismatch: `<=` requires a [number] or [datetime] on both sides but got `[number] <= [string]`
+
+### Boolean Evaluation With Warnings
+
+[Boolean Evaluation With Warnings]:#boolean-evaluation-with-warnings
+
+These are conditions that resolve to a boolean _and_ also record some warning information intended to help users understand why the evaluation happened the way it did. A classic example is using the `match` operator with a nil value on the left (this happens in the real world when the user uses a path on the left that doesn't match the given context). The `nil matches 'three'` expression returns a `false` response with the following warning message: "Type mismatch: `match` requires a non-nil value on the left"
+
+The `or`, `and`, and `not` operators will always return a boolean result _and_ will accumulate warnings from any evaluated sub-expressions.
+
+```json
+{
+  "a": {
+    "status": 503,
+    "env": "production"
+  }
+}
+```
+
+Expression | Evaluates like | Result | Accumulated Warnings
+-- | -- | -- | --
+`a.status >= 500` | `503 >= 500` | ✅ `true` | n/a
+`not a.expected == true` | `not (false w/ warning)` | ✅ `true` | Type mismatch: `==` requires the same type on both sides but got `[nil] == [bool]`
+`not a.expected == true and a.status >= 500` | `(not (false w/ warning)) and true`<br> -> `true and true` | ✅ `true` | Type mismatch: `==` requires the same type on both sides but got `[nil] == [bool]`
+`a.expected == true or a.status matches 'staging'` | `(false w/ warning) or (false w/ warning)`<br> -> `false or false` | ❌ `false` | (2 warnings)
+`not (a.expected == true or a.httpResponseCode < 500)` | `not((false w/ warning) or (non-boolean evaluation))`<br> -> `not(false or false)` | ✅ `true` | (2 warnings)
+
```

Comment:
**Not a blocker:** If you're accepting my other changes, above, where we remove links to the Note About Warnings... section, then we should remove that section entirely.

```suggestion
```

Reply:
Yeah, I had that question...